app/models/stripe/event_dispatch.rb

```diff
@@ -8,7 +8,7 @@ def dispatch_stripe_event(request)
     end
 
     def retrieve_stripe_event(request)
-      id = request['id']
+      id = request.params[:id]
```

Comment:
Hi @4nd2in can you split this change to it's own PR please? We might have to figure out a way to deal with this for rack versions that still use the old way.

Reply:
Hi, let me help.
stripe-rails supports `rails '>= 5.1'`
https://github.com/tansengming/stripe-rails/blob/master/stripe-rails.gemspec#L17

So rails 5.1.7 requires `rack '~> 2.0'`
https://github.com/rails/rails/blob/5-1-stable/actionpack/actionpack.gemspec#L29

I installed `rack 2.0.1` locally. Created simple app:
```
# config.ru

app = Proc.new do |env|
  request = Rack::Request.new(env)
  [200, {}, [request.params['id']]]
end

run app
```
Run `puma`
`http://localhost:9292/?id=12`
gives me correct result.

**Note that using a symbol for `id` does not work in this example!**

So older versions of rack support `request.params['id']`. We do not need to do anything.

Reply:
Created separate PR
https://github.com/tansengming/stripe-rails/pull/232